### PR TITLE
[DTensor][BE] improve DTensor ops correctness check utils

### PR DESCRIPTION
**Pull request by XilunWu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #158112
* #158051

**Summary**
Implemented the test pattern described in https://github.com/pytorch/pytorch/pull/157991#discussion_r2196363170 as a util method in `DTensorTestBase`. The difference to `DTensorTestBase._test_op` is:
1. allowing users to specify the `Partial` placement.
2. supporting tree-like output structure.

**Test**
so far only adopt `DTensorTestBase._test_op_on_dtensor` in `DistTensorOpsTest.test_split_on_partial`.
`pytest test/distributed/tensor/test_tensor_ops.py -s -k test_split_on_partial`

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @tianyu-l

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/testing/_internal/distributed/_tensor/common_dtensor.py [hunk 1]
```diff
@@ -403,6 +404,21 @@ def setUp(self) -> None:
         super().setUp()
         self._spawn_processes()
 
+    def _test_op_on_dtensor(self, dtensor: DTensor, op_call) -> None:
```
Comment (zpcore) [inline torch/testing/_internal/distributed/_tensor/common_dtensor.py]: This is super useful! Not necessary in this PR, is it possible we extend arg dtensor into *args, so that we can apply this function to different ops?

**Review by zpcore — APPROVED** [review]
LGTM!

File: torch/testing/_internal/distributed/_tensor/common_dtensor.py [hunk 2]
```diff
@@ -403,6 +404,21 @@ def setUp(self) -> None:
...
+        """
+        Unlike _test_op where the DTensor sharding is generated by DTensorConverter,
+        this function takes in DTensor object directly as argument and test the equality
+        of calling op on full_tensor() and DTensor.
+
+        Note: for in-place ops, the function may not work as expected if the DTensor
```
Comment (wconstab) [inline torch/testing/_internal/distributed/_tensor/common_dtensor.py]: shouldn't we be able to fix this limitation easily by just cloning the full tensor?

Comment (XilunWu) [reply torch/testing/_internal/distributed/_tensor/common_dtensor.py]: I'm not very sure that this is an actual problem. But I guess you're right, I should just preventively do `full_tensor().detach().clone()`

File: torch/testing/_internal/distributed/_tensor/common_dtensor.py (hunk 1 above)
Comment (XilunWu) [reply torch/testing/_internal/distributed/_tensor/common_dtensor.py]: yes, let me do it similarly as in `_test_op`.

**Comment by Skylion007** [comment]
Typo on git commit message though
